proxy/src/com/aerospike/client/proxy/BatchProxy.java

```diff
@@ -0,0 +1,138 @@
+import com.aerospike.client.policy.BatchPolicy;
+import com.aerospike.client.proxy.grpc.GrpcCallExecutor;
+import com.aerospike.client.util.Util;
+import com.aerospike.proxy.client.KVSGrpc;
+import com.aerospike.proxy.client.Kvs;
+import io.grpc.MethodDescriptor;
+
+/**
+ * All batch executors in one class mimicking
+ * {@link com.aerospike.client.async.AsyncBatch}.
+ */
+public class BatchProxy {
+    //-------------------------------------------------------
+    // OperateRecordSequence
+    //-------------------------------------------------------
+
+    // TODO @BrianNichols handle retries.
+    public static final class OperateRecordSequenceCommandProxy extends CommandProxy {
+        private final BatchRecordSequenceListener listener;
+        private final boolean[] sent;
+        private final BatchPolicy batchPolicy;
+        private final Key[] keys;
+        private final Operation[] ops;
+        private final BatchAttr attr;
+        private final boolean isOperation;
+
+        public OperateRecordSequenceCommandProxy(
+                GrpcCallExecutor grpcCallExecutor,
+                BatchPolicy batchPolicy,
+                Key[] keys,
+                Operation[] ops,
+                BatchRecordSequenceListener listener,
+                BatchAttr attr
+        ) {
+            super(grpcCallExecutor, batchPolicy);
+            this.batchPolicy = batchPolicy;
+            this.keys = keys;
+            this.ops = ops;
+            this.sent = new boolean[keys.length];
+            this.listener = listener;
+            this.attr = attr;
+            this.isOperation = ops != null;
+        }
+
+        @Override
+        protected MethodDescriptor<Kvs.AerospikeRequestPayload, Kvs.AerospikeResponsePayload> getGrpcMethod() {
+            return KVSGrpc.getBatchOperateStreamingMethod();
+        }
+
+        @Override
+        protected boolean isUnaryCall() {
+            return false;
+        }
+
+        @Override
+        void writeCommand(Command command) {
+            // The destination node is a single Aerospike proxy instance,
+            // where all keys are sent to the same node. Keys are not
+            // distributed across nodes.
+            // TODO @BrianNichols can a interface be implemented by both the
+            //  native and proxy client, to be passed to setBatchOperate as an
+            //  argument? node is passed in as "null" in BatchNode constructor.
+            BatchNode batchNode = new BatchNode(null, keys.length, 0);
+            for (int i = 1; i < keys.length; i++) {
+                batchNode.addKey(i);
+            }
+
+            command.setBatchOperate(batchPolicy, keys, batchNode, null, ops, attr);
+        }
+
+        @Override
+        void onResponse(Kvs.AerospikeResponsePayload response) {
+            byte[] bytes = response.getPayload().toByteArray();
+            Parser parser = new Parser(bytes, response.getStatus());
+
+            if (response.getHasNext()) {
```

Comment:
Since multiple batch requests are multiplexed on the same stream, the end of response is indicated by the "hasNext" flag.